tests/components/mqtt/test_common.py

```diff
@@ -1930,3 +1930,29 @@ async def help_test_discovery_setup(
     await hass.async_block_till_done()
     state = hass.states.get(f"{domain}.{name}")
     assert state and state.state is not None
+
+
+async def help_test_temperature_control_value_template_keys(
+    hass: HomeAssistant,
+    mqtt_mock_entry: MqttMockHAClientGenerator,
+    domain: str,
+    config: ConfigType,
+    attribute: str,
+    value_topic: str,
+    value_template: str,
+    payload: Any,
+) -> None:
+    """Test setting of attribute from an MQTT Temperature Control entity using templates."""
+    config = copy.deepcopy(config)
+    config[mqtt.DOMAIN][domain][value_topic] = "state-topic"
+    config[mqtt.DOMAIN][domain][value_template] = "{{ value_json }}"
+
+    await help_setup_component(
+        hass, mqtt_mock_entry, domain, config, use_discovery=True
+    )
+
+    async_fire_mqtt_message(hass, "state-topic", f'"{payload}"')
+    await hass.async_block_till_done()
+
+    state = hass.states.get(f"{domain}.test")
+    assert state.attributes.get(attribute) == payload
```

Comment:
Are we missing these tests for Climate?

Reply:
We have `test_climate.py -> test_get_with_templates`.
May be you could better keep this within `test_climate.py` and not place this in `test_common.py`.
Suggestion is to use `hass_config` or a discovery payload to set up the entity.

Reply:
I think I see where I've gone wrong there, it should have the same amount of coverage after removing the `help_test_temperature_control_value_template_keys` test now.